docs/backend-system/architecture/08-naming-patterns.md

```diff
@@ -12,10 +12,10 @@ As a rule, all names should be camel case, with the exceptions of plugin and mod
 
 ### Plugins
 
-| Description | Pattern           | Examples                              |
-| ----------- | ----------------- | ------------------------------------- |
-| export      | `<camelId>Plugin` | `catalogPlugin`, `userSettingsPlugin` |
-| ID          | `'<kebab-id>'`    | `'catalog'`, `'user-settings'`        |
+| Description | Pattern           | Examples                              | Notes                                                                 |
+| ----------- | ----------------- | ------------------------------------- | --------------------------------------------------------------------- |
+| export      | `<camelId>Plugin` | `catalogPlugin`, `userSettingsPlugin` |                                                                       |
+| ID          | `'<kebab-id>'`    | `'catalog'`, `'user-settings'`        | letters, digits, dashes, and underscores only, starting with a letter |
```

Comment:
Is this intentional? would this be valid `foo_bar-baz`? or would that be `foo-bar-baz`? Just wondering if we need the `notes` column or just saying use kebab-case is enough? WDYT?

Reply:
Im leaving it as it now, changing to kebab-case only seems more breaking. But happy to change it if we think we should be more restrictive

Reply:
Yep let's not allow underscores